benchmarks/twitter-finagle/src/main/scala/org/renaissance/twitter/finagle/FinagleHttp.scala

```diff
@@ -31,16 +32,38 @@ import org.renaissance.SimpleResult
 @Repetitions(12)
 class FinagleHttp extends RenaissanceBenchmark {
 
+  class WorkerThread(port: Int, barrier: CountDownLatch, requestCount: Int) extends Thread {
+    var totalContentLength = 0L
+
+    override def run(): Unit = {
+      val client: Service[http.Request, http.Response] =
+        com.twitter.finagle.Http.newService(s"localhost:$port")
+
+      barrier.countDown
+      barrier.await
+
+      for (i <- 0 until requestCount) {
+        val request = http.Request(http.Method.Get, "/json")
+        request.host = s"localhost:$port"
+        val response: Future[http.Response] = client(request)
+        Await.result(response.onSuccess { rep: http.Response =>
+          totalContentLength += rep.content.length
+        })
+      }
+      client.close()
+    }
+  }
+
   // TODO: Consolidate benchmark parameters across the suite.
   //  See: https://github.com/renaissance-benchmarks/renaissance/issues/27
 
   /** Number of requests sent during the execution of the benchmark.
    */
-  var NUM_REQUESTS = 2000000
+  var NUM_REQUESTS = 12000
 
   /** Number of clients that are simultaneously sending the requests.
    */
-  var NUM_CLIENTS = 20
+  var NUM_CLIENTS = 8
```

Comment:
Note that we have a consistency problem here. Here, the number of clients is just a constant, while in `finagle-chirper` the benchmark creates as many parallel clients as the number of available threads : 
https://github.com/renaissance-benchmarks/renaissance/blob/master/benchmarks/twitter-finagle/src/main/scala/org/renaissance/twitter/finagle/FinagleChirper.scala#L357

We definitely need some reasonable policy about parallelism to we could stick to...

Reply:
IMHO the approach in `finagle-chirper` is generally better. However, it would mean that all CPUs would be occupied by clients thus the CPUs would be shared with the server threads. Perhaps splitting the threads more evenly would make more sense?

Reply:
Yes, I think so too. Perhaps the benchmark should determine the parallelism level first:

```
val parallelism = Runtime.getRuntime.availableProcessors
```

And then split those among the agents of the benchmark. I'm wondering where the server thread count is set.

Reply:
Could we get this as part of this PR ? And maybe align `finagle-chirper` with the decision we will take here.